### PR TITLE
Add isTechnicalRoleAlias to EmailAddress

### PR DESCRIPTION
**Pull request by zackrowe**
This function will allow us to check for what we consider disallowed aliases that might be reported to watch dogs.

### CODE REVIEW

File: src/EmailAddress.php [hunk 1]
```diff
@@ -95,6 +104,16 @@ public function equals($other, bool $caseInsensitive = false)
         }
     }
 
+    /**
+     * Returns true if the email contains one of our defined disallowed aliases.
+     *
+     * @return bool
+     */
+    public function hasDisallowedAlias(): bool
```
Comment (ChrisSchwerdt) [inline src/EmailAddress.php]: These aren't inherently disallowed email addresses. They're just well known email aliases. It looks like there's even an [RFC](https://tools.ietf.org/html/rfc2142) listing them. Maybe rename this to `isStandardRoleAlias()`?

Comment (zackrowe) [reply src/EmailAddress.php]: Would we then want to expand the list to include all of the values in the RFC?

Comment (ChrisSchwerdt) [reply src/EmailAddress.php]: I think having `abuse`, `noc`, `security`, and `postmaster` makes the most sense. Maybe we could name it `isTechnicalRoleAlias()` in that case?

File: tests/EmailAddressTest.php [hunk 2]
```diff
@@ -140,4 +151,33 @@ public function equalityTestEmailAddresses(): array
...
+            [new EmailAddress('www@example.com'), false],
+            [new EmailAddress('uucp@example.com'), false],
+            [new EmailAddress('ftp@example.com'), false],
+            // Technical roles
+            [new EmailAddress('abuse@example.com'), true],
+            [new EmailAddress('noc@example.com'), true],
+            [new EmailAddress('security@example.com'), true],
+            [new EmailAddress('postmaster@example.com'), true],
+            // Test uppercase values.
+            [new EmailAddress('FTP@example.com'), true],
```
Comment (ChrisSchwerdt) [inline tests/EmailAddressTest.php]: Shouldn't this expect `false`? I'm guessing you had added `ftp` to the aliases and then removed it. We should probably just change it to `PostMaster@example.com`.

**Review by ChrisSchwerdt — APPROVED** [review]

**Pull request merged**